include/swift/AST/AutoDiff.h

```diff
@@ -422,7 +422,14 @@ class DerivativeFunctionTypeError
   Kind kind;
 
   /// The type and index of a differentiability parameter or result.
-  using TypeAndIndex = std::pair<Type, unsigned>;
+  /// std::pair does not have a trivial copy constructor on FreeBSD <= 14 for
```

Comment:
Could you just change the call sites that are using the constructor? That would simplify the diff.